js/ui/map.js

```diff
@@ -200,7 +200,9 @@ util.extend(Map.prototype, /** @lends Map.prototype */{
     addClass: function(klass, options) {
         if (this._classes[klass]) return;
         this._classes[klass] = true;
-        if (this.style) this.style._cascade(this._classes, options);
+        this._classOptions = options;
+        if (this.style) this.style.cascade();
+        this._update(true);
```

Comment:
I have a preference for moving logic out of `Map` and into `Style`. I wouldn't be opposed to adding more parameters to `_update` for the purpose.